File: the_niche_web/src/app/features/tenant-management/rooms/room.service.ts
import {Injectable} from "@angular/core";
import {HOUSE_API, ROOM_API} from "../../../shared/api-constants";
import {Observable} from "rxjs";
import {CrudInterface} from "../../../shared/service/crud.interface";
import {HttpClient} from "@angular/common/http";
import {RoomRequest, RoomResponse, RoomUpdateRequest} from "./room.model";


@Injectable({providedIn: 'root'})
export class RoomService implements CrudInterface<any>{

    constructor(private http:HttpClient) {
    }
    getAll(): Observable<any[]> {
        throw new Error("Method not implemented.");
    }
    getById(id: string | number): Observable<any> {
        throw new Error("Method not implemented.");
    }
    delete(id: string | number): Observable<void> {
        throw new Error("Method not implemented.");
    }
    getRoomsByHouse(id:string):Observable<RoomResponse[]>{
        return this.http.get<RoomResponse[]>(`${HOUSE_API}`+`/${id}/rooms`);
    }
    saveRoom(houseId:string,roomRequest:RoomRequest):Observable<RoomResponse>{
        const url = `${HOUSE_API}/${houseId}/rooms`;
        return this.http.post<RoomResponse>(url,roomRequest);
    }
    updateRoom(houseId:string,roomUpdateRequest:RoomUpdateRequest) :Observable<RoomResponse> {
        const url = `${HOUSE_API}/${houseId}/rooms/${roomUpdateRequest.id}`;
        return this.http.put<RoomResponse>(url,roomUpdateRequest);
    }
}

File: the-niche-coliving-api/src/main/java/com/theniche/colivin/room/RoomService.java
package com.theniche.colivin.room;
import com.theniche.colivin.room.dto.RoomRequest;
import com.theniche.colivin.room.dto.RoomResponse;
import java.util.List;
import java.util.UUID;

public interface RoomService {
    RoomResponse createRoom(UUID houseId, RoomRequest request);
    RoomResponse updateRoom(UUID houseId, UUID roomId, RoomRequest request);
    List<RoomResponse>  getRoomsByHouse(UUID houseId);
    // soft-delete
    void deleteRoom(UUID houseId, UUID roomId);

}
